### DISCUSSION TITLE
幽霊役職：能力仕様について

### DISCUSSION THREAD

**Discussion by yukieiji** [yukieiji/ExtremeRoles#45] [Ideas]
#  幽霊役職の能力仕様についての議論場所
大元のチケット：#44

- 新仕様
  - 能力のクールタイムは基本的に60秒、最小30～最大120秒で設定可
  - コミュサボ中は能力が全陣営一括で使用できない
  - パッシブの能力発動系は一切実装しない
  - 幽霊能力発動後の会議でどの役職がどんな能力を使ったのかを報告される
    - これは守護天使の物を応用

- 2022/05/17T23:55以前ディスカッションに記載されていた旧仕様
  - チケットに上がった段階ので仕様
    - コミュサボ中は能力が全陣営一括で使用できない
       - 基本的にボタンで発動する能力
       - 自動発動系の能力は無い(実装する意味があるのか？)
     - 能力のクールタイムは基本的に60秒
     - 今や今後実装されるEXロールの生存役職のインタフェースはどれも使用できない

**Comment by yukieiji** [comment]
生存役職役職のインタフェースが使えない場合以下のインタフェースが全て使えないことになる
- v2.0.5.1現在、EXロールに実装されているインタフェース
  - IRoleAbility：役職ボタン用の作成、呼び出し、リセット用のインターフェース
  - IRoleMurderPlayerHock：自分以外のキルが起きた時に何らかの処理をしたい場合のインターフェース
  - IRoleResetMeeting：会議が始まった、もしくは終わった時に何らかの処理をしたい場合のインターフェース
  - IRoleSpecialSetUp：役職が全員アサイン後のイントロ開始前、もしくは終わった時に何らかの初期化をしたい場合のインターフェース
  - IRoleUpdate：毎フレーム毎に何らかの処理をしたい場合のインターフェース
  - IRoleOption：役職のオプション用インターフェース(基本的に使わない(役職のベースクラスに内包されてる))
- v2.1.0.0実装予定のインタフェース
  - IRoleReportHock：通報時に何らかの処理をしたい場合のインタフェース
  - IRoleWinPlayerModifier：勝利条件等で勝利プレイヤー等を変更したい場合のインタフェース
  - IRoleSpecialReset：サイドキック等の処理で役職の内部状態全体をリセットする必要がある場合のインターフェース

**Comment by melonjagaimo** [comment]
幽霊だけの能力（CD60秒）ですか…
・他のプレイヤーに取り憑いて移動させる（移動以外の動作はとれる）
・他のプレイヤー1人に対し、自分の陣営を伝える
・他のプレイヤー1人の視界を広げる（TORのライターみたいな状態を付与）
といったものはどうですか？

**Reply by yukieiji** [reply]
幽霊役職の案は、そのベースとなる仕様が決まりきってないので能力の案を貰っても実装できる、出来ないってのは今判断できないです(それがバランス的にどうなのか？ってのも判断できないです)
この議論チケットは幽霊役職の能力の基礎仕様がこれで妥当かどうかを議論したい感じです

**Reply by melonjagaimo** [reply]
おっと、そうでしたか。それは失礼しました。
アクティブ能力のみにするのはいい考えだと思いますね。幽霊役職にパッシブ能力があっても最後の方に登場したら発動する間も無く終わってしまいますし

**Comment by yukieiji** [comment]
実装に先駆けて仕様書的なものを作るため
5月16日午前0時までに追加意見等がない場合は一旦この仕様での実装可能調査や詳しい内部仕様を決定します(決定後新しいディスカッションを作ります)

**Comment by yukieiji** [comment]
幽霊役職の仕様書を記載するため一時的にロックします

**Reply by yukieiji** [reply]
一時的な外部仕様が完成したためディスカッションを再度開放します
